### ISSUE TITLE
Expandir item nas tabelas

### ISSUE THREAD

**Issue by baraky** [fga-eps-mds/2021-2-SysArq-Doc#9]
## Descrição

Adicionar funcionalidade de expansão dos itens nas tabelas de listagem.

## Resultado esperado

Ao clicar no elemento, expandir para ter acesso a todos os dados cadastrados.

**Comment by guilherme-puida** [comment]
Com a implementação atual do componente das tabelas, implementar essa funcionalidade é um pouco mais difícil do que foi previsto. Os componentes individuais não tem conhecimento sobre o tipo de elemento que é contido na tabela, e sim somente a rota que corresponde ao tipo de elemento:

https://github.com/fga-eps-mds/2021.2-SysArq-Frontend/blob/a5064b656205729b52a5124763a168a27f6c5eb5/sysarq/src/pages/components/DataTable/index.js#L368-L383

Então para implementar essa funcionalidade, teríamos que criar novos componentes para as tabelas que saibam os elementos que elas contém, certo?

**Comment by baraky** [comment]
> 

Talvez não seja necessário criar um novo componente. Os componenetes de create (ex. [CreateAdministrativeProcess](https://github.com/fga-eps-mds/2021.2-SysArq-Frontend/blob/main/sysarq/src/pages/Documents/Create/CreateAdministrativeProcess.js)) provavelmente podem ser aproveitados. Mas realmente, a implementação será mais difícil do que o previsto.

Para os próximos passos, verifique se é realmente possível utilizar os componentes _create_ (provavelmente será preciso criar um novo argumento para mudar o formato do retorno).

**Comment by guilherme-puida** [comment]
Os componentes de create fazem muito mais que só mostrar os campos de cada item, como pode ser evidenciado pelos 50 useStates que existem no CreateAdministrativeProcess, por exemlo. Eu ponderei mais sobre o assunto e pensei que talvez seja possível passar um prop para cada DataTable que corresponda ao componente que deva ser mostrado quando a linha for expandida, mas acho que reutilizar os componentes de create não é a melhor solução.

**Comment by baraky** [comment]
Entendi, eu citei o create para reaproveitar código pois ele também é utilizado nas rotas ```/view/:id``` para visualizar as informações de um documento, apesar do nome não sugestivo. Mas sinta-se livre para outras possíveis soluções. Podemos discutir sobre na próxima reunião.

**Comment by guilherme-puida** [comment]
Sim, eu percebi que ele é usado nas rotas de visualiação também. Mas mesmo nessas rotas de visualização os campos são editáveis, ou eu estou enganado? Mesmo assim, acho que usar componentes mais simples que só mostrem as informações sem ter que lidar com nenhum tipo de estado interno seria uma solução mais adequada.

**Comment by guilherme-puida** [comment]
Vale a pena focar nessa funcionalidade agora ou focar nas US que tem no backlog? Não tive muito progresso essa semana.

**Comment by baraky** [comment]
> 

Pode trabalhar em outra issue, mas foque nas melhorias. Nosso compromisso é entregar as melhorias feitas quarta-feira.

Vamos ver de algum EPS trabalhar nessa issue aqui.

**Comment by baraky** [comment]
Movendo para Stand by até a refatoração do componente DataTable.

**Comment by baraky** [comment]
Issue recuperada. Foi acordado com o cliente que o melhor seria abrir o elemento numa modal.
